packages/ember-testing/lib/helpers.js

```diff
@@ -57,6 +57,8 @@ function focus(el) {
 
 function visit(app, url) {
   var router = app.__container__.lookup('router:main');
+  router._setupLocation();
```

Comment:
We should try to avoid manually calling `_setupLocation` here (as it forces the location to be ready before the actual application is fully booted).  However, this is precisely why the call to `router.location.setURL` was moved down in https://github.com/emberjs/ember.js/commit/8e130e52089d01d0110137bb67b11758d67e0f16 (because the location isn't ready yet).

Can you instead do something like the following (just after grabbing the `router:main` here):

```javascript
app.boot().then(function() {
  router.location.setURL(url);
});
```

Then remove the duplicate `router.location.setURL` call in the `else` block.